### PR TITLE
Fix epoch loss calculation in "NMT with attention" colab notebook

### PR DESCRIPTION
**Pull request by rakeshchada**
The epoch loss that's calculated in the current notebook is a bit inaccurate.

It is currently calculated as `total_loss/len(input_tensor)`. But the `total_loss` is already averaged across a batch. So, for `epoch loss` calculation, `total_loss` needs to be divided by the number of batches instead of total number of examples.

This PR adds code to calculate total number of batches `N_BATCH` and then calculate the epoch loss as `total_loss/N_BATCH`.

@lamberta @yashk2810 @tensorflower-gardener

### CODE REVIEW

File: tensorflow/contrib/eager/python/examples/nmt_with_attention/nmt_with_attention.ipynb [hunk 1]
```diff
@@ -677,21 +678,23 @@
...
         "        gradients = tape.gradient(loss, variables)\n",
-        "      \n",
+        "        \n",
         "        optimizer.apply_gradients(zip(gradients, variables), tf.train.get_or_create_global_step())\n",
-        "\n",
+        "        \n",
+        "        batch_loss += (loss / int(targ.shape[1]))\n",
+        "        \n",
+        "        total_loss += batch_loss\n",
+        "        \n",
```
Comment (yashk2810) [inline tensorflow/contrib/eager/python/examples/nmt_with_attention/nmt_with_attention.ipynb]: I don't think we need this because the batch loss printing that's happening indicates the loss of that particular batch.

File: tensorflow/contrib/eager/python/examples/nmt_with_attention/nmt_with_attention.ipynb [hunk 2]
```diff
@@ -677,21 +678,23 @@
...
         "        if batch % 100 == 0:\n",
         "            print('Epoch {} Batch {} Loss {:.4f}'.format(epoch + 1,\n",
         "                                                         batch,\n",
-        "                                                         loss.numpy() / int(targ.shape[1])))\n",
+        "                                                         batch_loss.numpy()))\n",
```
Comment (yashk2810) [inline tensorflow/contrib/eager/python/examples/nmt_with_attention/nmt_with_attention.ipynb]: Same here. I think this should remain the same.

File: tensorflow/contrib/eager/python/examples/nmt_with_attention/nmt_with_attention.ipynb [hunk 3]
```diff
@@ -677,21 +678,23 @@
...
         "    \n",
         "    print('Epoch {} Loss {:.4f}'.format(epoch + 1,\n",
-        "                                        total_loss/len(input_tensor)))\n",
+        "                                        total_loss/N_BATCH))\n",
```
Comment (yashk2810) [inline tensorflow/contrib/eager/python/examples/nmt_with_attention/nmt_with_attention.ipynb]: I agree with this change.

**Comment by yashk2810** [comment]
Just a few changes.

File: tensorflow/contrib/eager/python/examples/nmt_with_attention/nmt_with_attention.ipynb (hunk 1 above)
Comment (rakeshchada) [reply tensorflow/contrib/eager/python/examples/nmt_with_attention/nmt_with_attention.ipynb]: Ah it is the same functionality as before. I just assigned `loss/int(targ.shape[1]` to this variable `batch_loss` since it's being used in two places - once in `total_loss += loss/int(targ.shape[1]` and the other time during printing below. I meant to assign as `batch_loss = (loss / int(targ.shape[1]))` instead of doing `batch_loss += (loss / int(targ.shape[1]))`. So fixed that.

File: tensorflow/contrib/eager/python/examples/nmt_with_attention/nmt_with_attention.ipynb (hunk 2 above)
Comment (rakeshchada) [reply tensorflow/contrib/eager/python/examples/nmt_with_attention/nmt_with_attention.ipynb]: Ya `batch_loss` is the same thing as `loss/int(targ.shape[1])`.

File: tensorflow/contrib/eager/python/examples/nmt_with_attention/nmt_with_attention.ipynb (hunk 1 above)
Comment (rakeshchada) [reply tensorflow/contrib/eager/python/examples/nmt_with_attention/nmt_with_attention.ipynb]: Ah it is the same functionality as before. I just assigned loss/int(targ.shape[1] to this variable batch_loss since it's being used in two places - once in total_loss += loss/int(targ.shape[1] and the other time during printing below. I meant to assign as batch_loss = (loss / int(targ.shape[1])) instead of doing batch_loss += (loss / int(targ.shape[1])). So fixed that.

**Comment by yashk2810** [comment]
Can you please move the total_loss calculation above the variables = encoder.variables + decoder.variables line. It makes the flow of the training loop much clear even though its not used for loss calculation. After that LGTM.

File: tensorflow/contrib/eager/python/examples/nmt_with_attention/nmt_with_attention.ipynb (hunk 3 above)
Comment (yashk2810) [reply tensorflow/contrib/eager/python/examples/nmt_with_attention/nmt_with_attention.ipynb]: Can you please add spaces i.e. total_loss / N_BATCH

**Comment by yashk2810** [comment]
LGTM

**Comment by yashk2810** [comment]
@MarkDaoust @caisq Can you merge this PR?

**Review by MarkDaoust — APPROVED** [review]

**Comment by yashk2810** [comment]
@MarkDaoust I think you also need to add a "ready to pull" label.
